### PR TITLE
Implemented shared element transitions for addons

### PR DESCRIPTION
**Pull request by poisdeux**
Removed getting details in AddonDetailsFragment as the
AddonListFragment already provides all information required
in the AddonDetailsFragment. Therefore, we do not need to
retrieve the same information in the AddonDetailsFragment as well.

### CODE REVIEW

File: app/src/main/java/org/xbmc/kore/ui/AddonDetailsFragment.java [hunk 1]
```diff
@@ -225,70 +235,39 @@ public void onSuccess(String result) {
...
+        int posterHeight = resources.getDimensionPixelOffset(R.dimen.addondetail_poster_height);
+
+        UIUtils.loadImageIntoImageview(hostManager,
+                                       TextUtils.isEmpty(fanart)? poster : fanart,
+                                       mediaArt, artWidth, artHeight);
+        UIUtils.loadImageIntoImageview(hostManager,
+                                       poster,
+                                       mediaPoster, posterWidth, posterHeight);
 
-        setupEnableButton(addonDetails.enabled);
```
Comment (SyncedSynapse) [inline app/src/main/java/org/xbmc/kore/ui/AddonDetailsFragment.java]: You can't remove this call. It is used to show if the addon is enabled or not when the user enters the screen, so it is needed.

**Comment by poisdeux** [comment]
Oops... Fixed now. The "enabled" boolean now also gets passed from the list fragment to the details fragment.

File: app/src/main/java/org/xbmc/kore/ui/AddonDetailsFragment.java (hunk 1 above)
Comment (poisdeux) [inline app/src/main/java/org/xbmc/kore/ui/AddonDetailsFragment.java]: Moved it to the onCreateView method

**Comment by SyncedSynapse** [comment]
I'll merge it as is, as i want to build a release, but this introduces a small bug: Select an addon on the list, enabe/disabe it, return to the list, select the same addon, the state (enabled/disabled) of the addon will be incorrect because it is getting that state from the list which was updated before enabling/disabling it.

Keeping the GetAddonDetails call on the details fragment would prevent this.

**Comment by poisdeux** [comment]
So we only need to get the enabled state in the details fragment?

**Comment by SyncedSynapse** [comment]
yes.

**Pull request merged**